src/api/poolAPI.ts

```diff
@@ -546,18 +546,25 @@ export class PoolAPI extends BaseAPI {
 		referrer?: string,
 		taker?: string,
 		maxSlippagePercent?: number,
-		provider?: Provider
+		provider?: Provider,
+		poolKey?: PoolKey,
+		pool?: PoolMinimal
 	): Promise<FillableQuote> {
 		const _size = toBigInt(size)
 		const poolContract = this.premia.contracts.getPoolContract(
 			poolAddress,
 			provider ?? this.premia.multicallProvider
 		)
-		const [poolKey, quote, pool] = await Promise.all([
-			this.getPoolKeyFromAddress(poolAddress, provider),
+
+		const [quote, _poolKey, _pool] = await Promise.all([
 			poolContract.getQuoteAMM(taker ?? ZeroAddress, _size, isBuy),
-			this.getPoolMinimal(poolAddress),
+			(async () =>
+				poolKey
+					? poolKey
+					: await this.getPoolKeyFromAddress(poolAddress, provider))(),
+			(async () => (pool ? pool : await this.getPoolMinimal(poolAddress)))(),
```

Comment:
This is overly complex, can be simplified as suggested:

```suggestion
			poolKey ?? await this.getPoolKeyFromAddress(
			  poolAddress,
			  provider
			),
			pool ?? await this.getPoolMinimal(poolAddress),
```